src/planwise/component/importer.clj

```diff
@@ -89,18 +89,23 @@
         sites (:sites data)
         total-pages (:totalPages data)]
     (when (seq sites)
-      (let [new-facilities (sites->facilities sites type-field)]
+      (let [new-facilities (sites->facilities sites type-field)
+            sites-without-location (filter (complement with-location) sites)]
         (info (str "Dataset " dataset-id ": "
                    "Inserting " (count new-facilities) " facilities from page " page
                    " of collection " coll-id))
         (let [new-ids (facilities/insert-facilities! facilities dataset-id new-facilities)]
-          [:continue new-ids total-pages])))))
+          [:continue new-ids total-pages sites-without-location])))))
 
 (defn process-facilities
   [facilities facility-ids]
-  (doseq [id facility-ids]
-    (info "Processing facility" id)
-    (facilities/preprocess-isochrones facilities id)))
+  (map
+    (fn [id]
+      (info "Processing facility" id)
+      (facilities/preprocess-isochrones facilities id))
+    facility-ids))
```

Comment:
`map` returns a lazy sequence which may not be realised in the right moment, or the correct thread for that matter. Use `doall`, `mapv` or `(into [] (map ..))` to force immediate computation.